README.md

```diff
@@ -0,0 +1,131 @@
+office).
+
+There already are allocation scripts for various kinds of resources:
+
+- [resalloc-aws](https://github.com/praiskup/resalloc-aws) -
+  Allocates VMs in the Amazon EC2 cloud
+- [resalloc-ibm-cloud](https://github.com/fedora-copr/resalloc-ibm-cloud) -
+  Allocates VMs in the IBM Cloud
+- [resalloc-openstack](https://github.com/praiskup/resalloc-openstack) -
+  Allocates VMs in your OpenStack instance
+- [resalloc-kubernetes](https://github.com/TommyLike/resalloc-kubernetes) -
+  Allocates pods in your Kubernetes cluster
+
+If you can, please share your own.
+
+
+Motivation
+----------
+
+At the first sight, it may look like resource allocation is a simple task
+and using something like resalloc is an overkill. But please consider what
+resalloc can do for you and what you would eventually need to implement on your
+own when starting from scratch.
+
+- Multiple resource pools - Resalloc allows to define multiple pools of
+  resources. They can either provide identical resources or different
+  kinds of resources (e.g. x86_64 VMs from Amazon AWS and x86_64 VMs from
+  OpenStack that can be used interchangeably, and ppc64le VMs from IBM Cloud
+  that are used for a different purpose).
+- Client-server architecture - Allocating many new resources in parallel can be
+  surprisingly heavy on CPU and memory, so it can be useful to dedicate a
+  separate machine to resource allocation.
+- Gradual allocation - Allocating resources on self-hosted hardware is for free,
+  so it makes sense to use it to the full capacity at all times. However,
+  somebody will have to pay for every allocated resource in the cloud, so we
+  want to allocate only as much resources as we need at the moment.
+- Preallocation - Allocating new resources can take some time (e.g. spawning a
+  new VM in the cloud and running ansible playbooks to provision it can take few
+  minutes), and users don't want to wait. It is a good idea to preallocate a
+  small number of resources that are ready to be used immediately.
+- Livechecks - Clouds are unreliable. VMs can break while starting or become
+  unresponsive for various reasons. Resalloc periodically checks the livelines
+  of all resources and makes sure money doesn't leak out of our pockets.
+- Resource prioritization - Multiple pools can provide the same resources but
+  they might not cost the same. There is a price for running VMs in Amazon
+  AWS. The same VMs as Spot instances are a bit cheaper, and running them on a
+  self-hosted sever is for free. Resalloc prioritizes to be wallet-friendly.
+- Web interface - It is possible to optionally run a web UI and let users see
+  what, and how many resources are available. For example, see
+  [Copr resources](https://download.copr.fedorainfracloud.org/resalloc).
+
+Resalloc was created to accomodate the ever growing
+[farm of Copr builders](https://pavel.raiskup.cz/blog/copr-farm-of-builders.html).
```

Comment:
To be even more precise, the history is much longer - I used to have a shell-only solution (I don't think it was ever published).  The motivation was to have the same VM allocator tool both for Copr (the Red Hat) instance and the CI (Jenkins based) for the https://github.com/sclorg images (the team doesn't use resalloc anymore).